docs/design/networking.md

```diff
@@ -97,11 +99,18 @@ So we end up with 3 cases:
 
 1. Container -> Container or Container <-> VM. These should use 10. addresses directly and there should be no NAT.
 
-2. Container -> Internet. These have to get mapped to the primary host IP so that GCE knows how to egress that traffic. There is actually 2 layers of NAT here: Container IP -> Internal Host IP -> External Host IP. The first level happens in the guest with IP tables and the second happens as part of GCE networking. The first one (Container IP -> internal host IP) does dynamic port allocation while the second maps ports 1:1.
+2. Container -> Internet. These have to get mapped to the primary host IP so that the host node knows how to egress that traffic. In most networking setups there are actually 2 layers of NAT here: Container IP -> Internal Host IP -> External Host IP. The first level happens in the guest with IP tables and the second happens as part of egress past a firewall to the broader internet.  On cloud providers, this is usually provided by the cloud virtual network.  On physical providers, this is provided by some sort of gateway router.. The first one (Container IP -> internal host IP) does dynamic port allocation while the second maps ports 1:1.
```

Comment:
I'm not going to pick nits about being too GCE-centric - this doc needs an overhaul from that perspective.

"router.." - double dot

I think the last sentence here is more confusing than helpful.  Move to strike the whole sentence.  Specifically, outbound ports almost never matter, and I have not nearly enough breadth to endorse any statement of how these things are usually done outside of the core linux kernel implementation.